### PR TITLE
Fix links for Title IX and Maintainer Notes

### LINKED ISSUE #2012: Fix broken Title IX link + maintainer-notes.html
[link check failed](https://github.com/opendp/opendp/actions/runs/10649348848/job/29519450979)
```
URL        `https://docs.opendp.org/en/stable/contributing/maintainer-notes.html'
Parent URL file:///home/runner/work/opendp/opendp/.github/workflows/README.md, line 7, col 23
Real URL   https://docs.opendp.org/en/stable/contributing/maintainer-notes.html
Check time 1.734 seconds
Size       821B
Result     Error: 404 Not Found
 5 threads active,     0 links queued,   62 links in  70 URLs checked, runtime 6 seconds

URL        `https://www.titleix.harvard.edu/'
Parent URL file:///home/runner/work/opendp/opendp/CODE_OF_CONDUCT.md, line 15, col 225
Real URL   https://www.titleix.harvard.edu/
Check time 1.036 seconds
Size       970B
Result     Error: 404 Not Found
```
- There doesn't seem to be a top level Title IX site any more. Replace with SEAS?
- Maintainer notes were outdated and we dropped them.

### PR DESCRIPTION
**Pull request by mccalluc**
- Fix #2012

Got approval from Sharon on Slack.

### CODE REVIEW

File: .github/workflows/README.md [hunk 1]
```diff
@@ -4,8 +4,8 @@
 
 OpenDP uses a number of Github workflows: Some are started automaticly
 (pushes, PRs, and scheduled runs), and others are manual (via the github UI or API).
-The [Maintainer Notes](https://docs.opendp.org/en/stable/contributing/maintainer-notes.html)
-provide a broader view of the the develpment and release process.
+The [Development Environment](https://docs.opendp.org/en/stable/contributing/development-environment.html) section of the docs
+provides a broader view of the the develpment and release process.
```
Comment (Shoeboxam) [inline .github/workflows/README.md:8]: I think I'd just drop this, since we dropped those docs.

File: .github/workflows/README.md [hunk 2]
```diff
@@ -4,8 +4,8 @@
 
 OpenDP uses a number of Github workflows: Some are started automaticly
```
Comment (Shoeboxam) [inline .github/workflows/README.md:5]:
```suggestion
OpenDP uses a number of Github workflows: Some are started automatically
```

**Review by Shoeboxam — CHANGES_REQUESTED** [review]
From Chuck, Sharon gave thumbs up on pointing to SEAS.

File: .github/workflows/README.md (hunk 1 above)
Comment (mccalluc) [inline .github/workflows/README.md:8]:
```suggestion
```

**Review by Shoeboxam — APPROVED** [review]
Thanks for the edits!

**Pull request merged**